### DISCUSSION TITLE
How to set the cursor position after applying a template?

### DISCUSSION THREAD

**Discussion by daamiian** [SilentVoid13/Templater#673] [Help]
Is there a function which will allow me to choose where in the template to place the cursor for typing after it's been applied?

This is what I'm trying to do:

TAG HERE `<span style='background-color: #ff461e; color: #f7f3ee; border-top-left-radius: 5px; border-bottom-left-radius: 5px; border-top-right-radius: 5px; border-bottom-right-radius: 5px; padding: 1'> <b><i> TEXT HERE <i><b></span>`

Basically I want an emoji tag (TAG HERE) to follow whatever text I want highlighted in red (I'm using this as a bookmark within notes - so I'll apply the template when I'm ready to type a note for my bookmark, indicated by an emoji). I want to apply the template, then it automatically moves the cursor to the TEXT HERE position ready to type so I don't have to do anything other than type once I apply the template. Can someone help me out? I tried tp.file.cursor() but can't get it to work. Thanks in advance!

**Comment by welpdx** [comment]
> I tried tp.file.cursor() but can't get it to work

When using `tp.file.cursor()`, make sure that you have `Automatic jump to cursor` enabled in Templater Settings

Edit: ![img](https://i.imgur.com/br04mXk.jpeg)

**Reply by daamiian** [reply]
@welpdx yes unfortunately I have this enabled already. When I use tp.file.cursor(), there is no focus on the document at all. 

In fact, I just did a check on a basic template and tp.file.cursor() doesn't seem to auto jump no matter what, even with the option enabled. Maybe there is a plugin confliction with andy mode, my best guess. I tried the following template:

`# <% tp.file.cursor() %>`

in order to check if the cursor jumped to the position after the heading or not, and it didn't. No focus anywhere. Not after I press enter, tab, ctrl+tab, nothing. I have to physically click on my pane to type.

**Reply by welpdx** [reply]
ok maybe it is about #583 or #634
```js
<%* app.workspace.activeLeaf.view.editor.focus() %>
```
